File: day:52/invertBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if(root is None): return root;
    
        queue = [root];
    
        while(len(queue) > 0):
            node = queue.pop(0);
            temp = node.left;
            node.left = node.right;
            node.right = temp;
            if(node.left):
                queue.append(node.left);
            if(node.right):
                queue.append(node.right);
        return root;

File: day:52/invertBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    if(root === null) return root;
    
    let queue = [root];
    
    while(queue.length > 0) {
        let node = queue.shift();
        let temp = node.left;
        node.left = node.right;
        node.right = temp;
        if(node.left) {
            queue.push(node.left);
        }
        if(node.right) {
            queue.push(node.right);
        }
    }
    return root;
};
